controllers/helmchart_controller.go

```diff
@@ -390,10 +390,10 @@ func (r *HelmChartReconciler) reconcileSource(ctx context.Context, obj *sourcev1
 		// Return Kubernetes client errors, but ignore others which can only be
 		// solved by a change in generation
 		if apierrs.ReasonForError(err) == metav1.StatusReasonUnknown {
-			return sreconcile.ResultEmpty, &serror.Stalling{
-				Err:    fmt.Errorf("failed to get source: %w", err),
-				Reason: "UnsupportedSourceKind",
-			}
+			return sreconcile.ResultEmpty, serror.NewStalling(
```

Comment:
Because this reconciler still uses the old `RecordContextualError` for result processing, this need not use the constructor yet. Let's undo this change here and in helmrepository reconciler.
Only gitrepository and ocirepository reconcilers use the new `ErrorActionHandler` that benefits from using this constructor. It's been a long time since `ErrorActionHandler` is in use and we haven't seen any issues with it. We should be able to update the rest of the reconcilers to use it. Better done in a separate PR.